NuRadioReco/modules/io/coreas/readCoREASShower.py

```diff
@@ -0,0 +1,160 @@
+            # if(filesize < 18456 * 2):
+            #     logger.warning("file {} seems to be corrupt, skipping to next file".format(
+            #            self.__input_files[self.__current_input_file]))
+            #     self.__current_input_file += 1
+            #     continue
+
+            if self.__verbose:
+                print('Reading %s ...' % self.__input_files[self.__current_input_file])
+
+            corsika = h5py.File(self.__input_files[self.__current_input_file], "r")
+            logger.info("using coreas simulation {} with E={:2g} theta = {:.0f}".format(
+                self.__input_files[self.__current_input_file], corsika['inputs'].attrs["ERANGE"][0] * units.GeV,
+                corsika['inputs'].attrs["THETAP"][0]))
+
+            f_coreas = corsika["CoREAS"]
+
+            evt = NuRadioReco.framework.event.Event(corsika['inputs'].attrs['RUNNR'], corsika['inputs'].attrs['EVTNR'])
+            evt.__event_time = f_coreas.attrs["GPSSecs"]
+
+            sim_shower = coreas.make_sim_shower(corsika)
+            sim_shower.set_parameter(shP.core, np.array([0, 0, f_coreas.attrs["CoreCoordinateVertical"] / 100]))  # overwrite core
+            evt.add_sim_shower(sim_shower)
+
+            if self.__highlevel:
+                try:
+                    f_highlevel = corsika["highlevel"]
+                except KeyError:
+                    logger.KeyError("No highlevel group")
+                    raise ValueError("No highlevel group")
+
+                # choose first one, rest is read latter
+                planes = list(f_highlevel.keys())
+                obs_plane = f_highlevel[planes[0]]
+
+                # in case simulation is not a star shaped
+                try:
+                    # evt.set_parameter("radiation_energy_1D", obs_plane.attrs["radiation_energy_1D"] * units.eV)
+                    evt.set_parameter(shP.radiation_energy, obs_plane.attrs["radiation_energy"] * units.eV)
+                except KeyError:
+                    pass
+
+                antenna_position = obs_plane["antenna_position"]
+                antenna_position_vBvvB = obs_plane["antenna_position_vBvvB"]
+                energy_fluence = obs_plane["energy_fluence"]
+                energy_fluence_vector = obs_plane["energy_fluence_vector"]
+                frequency_slope = obs_plane["frequency_slope"]
+                names = obs_plane["antenna_names"]
+
+                # later parse station id from antenna name if possible
+                for idx, pos in enumerate(np.array(antenna_position)):
+
+                    station_id = antenna_id(names[idx], idx)  # return proper station id if possible
+                    station = NuRadioReco.framework.station.Station(station_id)
+
+                    sim_station = NuRadioReco.framework.sim_station.SimStation(station_id, position=pos)
+                    electric_field = NuRadioReco.framework.electric_field.ElectricField([0, 1, 2])
```

Comment:
The channel IDs should not be hardcoded. Use the channel IDs from the detector description instead.